### PR TITLE
Release Data.Array.Generic.Mut

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: core/src/Streamly/Data/Array/Generic/Mut.hs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+-- License     : BSD3-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+module Streamly.Data.Array.Generic.Mut
+(
+    -- * Type
+    -- $arrayNotes
+      Array (..)
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Array/Generic/Mut.hs:13]: Do not expose the constructor.

File: core/src/Streamly/Data/Array/Generic/Mut.hs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    -- * Constructing and Writing
+    -- ** Construction
+    -- *** Uninitialized Arrays
+    , newArray
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Array/Generic/Mut.hs:18]: The names in this module should be consistent with names in Streamly.Data.Array/Streamly.Data.Array.Mut .

File: core/src/Streamly/Data/Array/Generic/Mut.hs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    -- *** From streams
+    , writeNUnsafe
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Array/Generic/Mut.hs:21]: Do not expose everything from the internal module. Please see what is exposed from other array modules and expose only carefully reviewed APIs that are consistent across all array modules.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: core/src/Streamly/Data/Array/Generic/Mut.hs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    -- bounds in the array representation.
+
+    -- ** Appending elements
+    , snoc
+
+    -- * Eliminating
+    , toList
+
+    -- ** Random reads
+    , getIndex
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Array/Generic/Mut.hs:38]: Not consistent with the API in Data.Array.Mut.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Can we keep the haddock headings consistent with Data.Array.Mut?
Export reader.

**Comment by rnjtranjan** [comment]
Fix:https://github.com/composewell/streamly/issues/2083

**Pull request merged**